### DISCUSSION TITLE
Running MCMC with Pastas model

### DISCUSSION THREAD

**Discussion by IsaDio** [pastas/pastas#347] [Q&A] [answered]
Greetings!

Thanks for including a tool like MCMC to assess uncertainty (non-uniqueness of solution) in Pastas.
I have been applying MCMC to a model and I get satisfactory results, but I have a few questions/comments.
Sorry if this is a long post but I thought it might be worth sharing my experience.

- I first calibrate my model using the standard method and then solve again using the emcee method without initialising the parameters. Like in your example (from Pastas documentation) I would expect the solution with emcee to be similar to my first solution. It is mostly the case except emcee always moves my constant_d variable up a little, which doesn't give such great results if compared to my observations. Do you have an idea why it does that? Is it because my calibrated constant_d is close to its lower bound (I fix an upper and lower bounds to control a bit the solution based on knowledge). At the moment I mostly ignore the blue line.
First image below is my standard calibration and second figure shows my MCMC results with the calibrated solution in blue.
![image](https://user-images.githubusercontent.com/91039837/143148514-ff135673-596c-4ee8-9aa7-6548bee11ad8.png)
![image](https://user-images.githubusercontent.com/91039837/143148888-e0bdf3fe-5ea3-46a9-bd5e-7e48782d2034.png)

- This is something I found while running emcee but it only generates parameters for the ones not fixed. This means if I am using the recharge FlexModel I need to add the 3 constants (recharge_lp, recharge_simax and recharge_kv) to the array before the simulation like the below.
![image](https://user-images.githubusercontent.com/91039837/143149685-52e21cd5-8bb4-4a57-b56d-7dc73f28d8de.png)

- I found that the range of results produced by MCMC was a bit wide based on my range of observations so I added the option to select the member based on its RMSE, which gives me the spread on the figure above for example. This I think is really useful.

**Comment by raoulcollenteur** [comment, marked as answer]
Hi @IsaDio,

Thanks for sharing the example with us! Using MCMC can be quite challenging, and often you need quite a lot of steps for the walkers to get good (and stable) results. I think then Emcee package documentation has quite good examples on this that may help as well (https://emcee.readthedocs.io/en/stable/).

Did you try fixing the constant_d parameter? That may help a bit. Or perhaps running longer chains already solves the issue? Otherwise you could try giving more informative priors, this may actually help a lot for the non-linear recharge model.

Cheers,
Raoul
